src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -3062,5 +3062,35 @@ unsafe static class C
                 Diagnostic(ErrorCode.WRN_UnreferencedField, "Ptr").WithArguments("C.Ptr").WithLocation(5, 65)
             );
         }
+
+        [Fact, WorkItem(46688, "https://github.com/dotnet/roslyn/issues/46688")]
+        public void NewAfterPtrDeclaration()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    void M1()
+    {
+        delegate*<void> ptr = new () => {};
+    }
+
+    void M2()
+    {
+        delegate*<void> ptr = new();
```

Comment:
>new(); [](start = 30, length = 6)

If we don't have one already, could you include an object creation test with explicit type as well?